web/src/user_group_pill.ts

```diff
@@ -47,32 +58,41 @@ export function get_user_ids(pill_widget) {
     return user_ids;
 }
 
-export function append_user_group(group, pill_widget) {
+type PillWidget = {
+    appendValidatedData: (item: PillItem) => void;
+    clear_text: () => void;
+    items: () => PillItem[];
+};
```

Comment:
This type belongs to `input_pill.js` as the return type of `input_pill.create`. Until `input_pill.js` is migrated, we should put it in `types.ts` with a TODO comment like the other types there.